src/main/java/de/tum/in/www1/artemis/service/hestia/ProgrammingExerciseTaskService.java

```diff
@@ -184,13 +223,79 @@ private List<ProgrammingExerciseTask> extractTasks(ProgrammingExercise exercise)
             var testCaseNames = extractTestCaseNames(capturedTestCaseNames);
 
             for (String testName : testCaseNames) {
-                testCases.stream().filter(tc -> tc.getTestName().equals(testName)).findFirst().ifPresent(task.getTestCases()::add);
+                Optional<ProgrammingExerciseTestCase> foundTestCase;
+                if (testName.startsWith(TESTID_START)) {
+                    Long id = extractTestId(testName);
+                    foundTestCase = testCases.stream().filter(tc -> tc.getId().equals(id)).findFirst();
+                }
+                else {
+                    foundTestCase = testCases.stream().filter(tc -> tc.getTestName().equals(testName)).findFirst();
+                }
+                foundTestCase.ifPresent(task.getTestCases()::add);
             }
             tasks.add(task);
         }
         return tasks;
     }
 
+    /**
+     * Replaces a comma seperated list of test case names with their corresponding ids.
+     * If no matching test case exists (e.g. due to a typo), we keep the test name.
+     */
+    private String extractTestCaseIdsFromNames(String capturedTestCaseNames, Set<ProgrammingExerciseTestCase> testCases) {
+        var testCaseNames = extractTestCaseNames(capturedTestCaseNames);
+
+        return testCaseNames.stream().map(testName -> convertTestNameToTestId(testName, testCases)).collect(Collectors.joining(","));
+    }
+
+    private String convertTestNameToTestId(String testName, Set<ProgrammingExerciseTestCase> testCases) {
+        for (ProgrammingExerciseTestCase testCase : testCases) {
+            if (testName.equals(testCase.getTestName())) {
+                String id = testCase.getId().toString();
+                return TESTID_START + id + TESTID_END;
+            }
+        }
+        return testName;
+    }
+
+    private String extractTestNamesFromIds(String capturedTestCaseIds, Set<ProgrammingExerciseTestCase> testCases) {
+        var capturedTestIds = extractTestCaseNames(capturedTestCaseIds);
+
+        return capturedTestIds.stream().map(tc -> convertTestIdToTestName(tc, testCases)).collect(Collectors.joining(","));
+    }
+
+    private String convertTestIdToTestName(String testId, Set<ProgrammingExerciseTestCase> testCases) {
+        Long id = extractTestId(testId);
+
+        if (id == null) {
+            // no matching test case e.d. due to a typo, leave it as it is
+            return testId;
+        }
+
+        for (ProgrammingExerciseTestCase tc : testCases) {
+            if (tc.getId().equals(id)) {
+                String testName = tc.getTestName();
+                return Objects.requireNonNullElse(testName, testId);
+            }
+        }
+        return testId;
+    }
+
+    private Long extractTestId(String test) {
```

Comment:
About the naming of test vs. testID: Here, you call a string like `<testid>123</testid>` simply `test` 🤔 

Also, can you Add an example of what this method does in a JavaDoc comment? Something like "example: <testid>123</testid> -> 123" suffices to understand the method much more quickly.